src/parsing.jl

```diff
@@ -32,125 +51,102 @@ in Section 9.7 of the IEEE Std 1788-2015, with some extensions.
 
 The created interval is tight around the value described by the string,
 including for number that have no exact float representation like "0.1".
-
-Roughly speaking, the valid forms are
-    - `[ 1.33 ]` or simply `1.33` : The interval containing only `1.33``.
-    - `[ 1.44, 2.78 ]` : The interval `[1.44, 2.78]`.
-    - `7.88 ± 0.03`: The interval `[7.85, 7.91]`.j
-    - `6.42?2` : The interval `6.42 ± 0.02`. The number after `?` represent
-        the uncertainty in the last digit.
-        Physicists would write it as `6.42(2)`.
-        Strangely enough, according to the standard, the default
-        value is `0.5` (e.g. `2.3? == 2.3 ± 0.05`).
-        The direction of the uncertainty can be given by adding 'u' or 'd' at
-        the end for the error going only up or down respectively (e.g.
-        `4.5?5u == [4.5, 5]`).
 """
 function parse(::Type{Interval{T}}, s::AbstractString) where T
-    p = interval_parser(Interval{T})
+    s = lowercase(strip(s))
     try
-        return parse(p, s)
-    catch e
-        # We avoid CombinedParsers stacktraces because they are hard to read.
-        if !(e isa ArgumentError)
-            rethrow()
-        end
+        ival, _ = _parse(Interval{T}, s)
+        return ival
+    catch
+        @warn "invalid input, empty interval returned"
+        return emptyinterval(T)
     end
+end
 
-    throw(ArgumentError("string \"$s\" can not be parsed to an interval."))
+function _parse(::Type{Interval{T}}, s::AbstractString) where T
```

Comment:
Documenting here (e.g. in a docstring) what `isnotcom` do would be nice.

Also we could document (either here or in the docstring of `parse`) that `parse(Intrerval, s)` never errors, but `_parse(Interval, s)` does. The built-in `parse`, e.g. `parse(Int, "a")`, does error on invalid input, so we are doing something a bit special here.